bundles/org.openhab.binding.elgatokeylight/src/main/java/org/openhab/binding/elgatokeylight/internal/ElgatoKeyLightHandler.java

```diff
@@ -0,0 +1,181 @@
+        ElgatoKeyLight light = this.light;
+        if (light == null) {
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.OFFLINE.NONE, "Not initialized!");
+            return;
+        }
+
+        try {
+            LightStatus lightStatus = light.readLightStatus();
+
+            if (command instanceof RefreshType) {
+                updateState(lightStatus);
+                return;
+            }
+
+            switch (channelUID.getId()) {
+            case CHANNEL_BRIGHTNESS:
+                if (command instanceof PercentType) {
+                    boolean on = !PercentType.ZERO.equals(command);
+                    int brightness = ((PercentType) command).intValue();
+                    lightStatus = light.switchTo(on, brightness);
+                } else if (command instanceof OnOffType) {
+                    boolean on = OnOffType.ON.equals(command);
+                    lightStatus = light.switchTo(on, lightStatus.brightness > 0 ? lightStatus.brightness : 100);
+                } else if (command instanceof IncreaseDecreaseType) {
+                    int brightness = LightStateConverter.toAdjustedBrightness((IncreaseDecreaseType) command,
+                            lightStatus.brightness);
+                    boolean on = brightness > 0;
+                    lightStatus = light.switchTo(on, brightness);
+                }
+                updateState(lightStatus);
+                break;
+
+            case CHANNEL_COLOR_TEMPREATURE:
+                if (command instanceof PercentType) {
+                    int temperature = LightStateConverter.toColorTemperatureLightState((PercentType) command);
+                    lightStatus = light.writeTemperature(temperature);
+                } else if (command instanceof OnOffType) {
+                    boolean on = OnOffType.ON.equals(command);
+                    lightStatus = light.switchTo(on, lightStatus.brightness > 0 ? lightStatus.brightness : 100);
+                } else if (command instanceof IncreaseDecreaseType) {
+                    int temperature = LightStateConverter.toAdjustedColorTemp((IncreaseDecreaseType) command,
+                            lightStatus.brightness);
+                    lightStatus = light.writeTemperature(temperature);
+                }
+                updateState(lightStatus);
+                break;
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+            return;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(ElgatoKeyLightConfiguration.class);
+
+        // set the thing status to UNKNOWN temporarily and let the background task
+        // decide for the real status.
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            if ((config.host == null) || config.host.isEmpty()) {
```

Comment:
The compiler warnings can be solved by storing `config` to a local variable and do the null check on that.